libp2p-ping/src/lib.rs

```diff
@@ -156,8 +159,19 @@ impl<C> ConnectionUpgrade<C> for Ping
 		let sink_stream = socket.framed(Codec).map(|msg| Message::Received(msg.freeze()));
 		let (sink, stream) = sink_stream.split();
 
+		let remote_addr = if log_enabled!(target: "libp2p-ping", Level::Debug) {
+			Some(remote_addr.clone())
+		} else {
+			None
+		};
+
 		let future = loop_fn((sink, stream.select(rx)), move |(sink, stream)| {
 			let expected_pongs = expected_pongs.clone();
+			let remote_addr = if log_enabled!(target: "libp2p-ping", Level::Debug) {
+				Some(remote_addr.clone())
```

Comment:
isn't the result of this operation `Option<Option<Multiaddr>>`? Is it intended?

Reply:
Ah, good catch.